src/Entities/User.php

```diff
@@ -124,6 +128,51 @@ public function getEmailIdentity(): ?UserIdentity
         return $this->getIdentity('email_password');
     }
 
+    /**
+     * If $user, $password, or $password_hash have been updated,
+     * will update the user's email identity record with the
+     * correct values.
+     */
+    public function saveEmailIdentity(): bool
+    {
+        if (empty($this->email) && empty($this->password) && empty($this->password_hash)) {
+            return true;
+        }
+
+        $identity = $this->getEmailIdentity();
+        if ($identity === null) {
+            // Ensure we reload all identities
+            $this->identities = null;
+
+            $this->createEmailIdentity([
+                'email'    => $this->email,
+                'password' => '',
+            ]);
+            $identity = $this->getEmailIdentity();
+        }
+
+        if (empty($identity)) {
+            return true;
+        }
```

Comment:
It seems `$identity` is never empty here. When is it empty?

Reply:
Good catch. changed.